### PR TITLE
[CRC] add docstring for opcode executor

### PR DESCRIPTION
**Pull request by zrr1999**

### CODE REVIEW

File: sot/opcode_translator/executor/opcode_executor.py [hunk 1]
```diff
@@ -205,9 +283,18 @@ def inner(self: OpcodeExecutorBase, instr: Instruction):
 
 
 def jump_break_graph_decorator(normal_jump):
-    """breakoff graph when meet jump."""
+    """
+    A decorator function that breaks off the graph when a jump instruction is encountered.
```
Comment (Aurelius84) [inline sot/opcode_translator/executor/opcode_executor.py]:
```suggestion
    A decorator function that breaks off the graph when a JUMP-related instruction is encountered.
```

File: sot/opcode_translator/executor/opcode_executor.py [hunk 2]
```diff
@@ -221,10 +308,21 @@ def inner(self: OpcodeExecutor, instr):
     return inner
 
 
-def call_break_graph_decorator(push_n):
-    def decorate(call_fn):
+def call_break_graph_decorator(push_n: int):
+    """
+    A decorator function that breaks off the graph when a function call instruction is encountered.
```
Comment (Aurelius84) [inline sot/opcode_translator/executor/opcode_executor.py]:
```suggestion
    A decorator function that breaks off the graph when a function CALL instruction is encountered.
```

File: sot/opcode_translator/executor/opcode_executor.py [hunk 3]
```diff
@@ -251,6 +360,35 @@ def inner(*args, **kwargs):
 
 
 class OpcodeExecutorBase:
+    """
+    Base class for executing opcode instructions.
+
+    The OpcodeExecutorBase class provides methods and functionality to execute opcode instructions.
+
+    If you want to learn more about Python instructions, you can visit https://docs.python.org/3/library/dis.html
```
Comment (Aurelius84) [inline sot/opcode_translator/executor/opcode_executor.py]:
```suggestion
    If you want to learn more about Python instructions, see https://docs.python.org/3/library/dis.html for details.
```

File: sot/opcode_translator/executor/opcode_executor.py [hunk 4]
```diff
@@ -265,28 +403,74 @@ def __init__(self, code: types.CodeType, graph: FunctionGraph):
...
+
+        Raises:
+            NotImplementedError: If the method is not implemented.
+
+        """
         raise NotImplementedError()
 
     def transform(self):
+        """
+        Transforms the executor.
```
Comment (Aurelius84) [inline sot/opcode_translator/executor/opcode_executor.py]:
```suggestion
        Abstract method need to be implemented to symbolic translate each instruction.
```

File: sot/opcode_translator/executor/opcode_executor.py [hunk 5]
```diff
@@ -1167,7 +1467,18 @@ def _break_graph_in_jump(self, result, instr):
         self.guard_fn = self._graph.guard_fn
 
     @fallback_when_occur_error
-    def _break_graph_in_call(self, origin_stack, instr, push_n):
+    def _break_graph_in_call(
+        self, origin_stack: list[VariableBase], instr: Instruction, push_n: int
+    ):
+        """
+        Break the graph at a call instruction.
```
Comment (Aurelius84) [inline sot/opcode_translator/executor/opcode_executor.py]:
```suggestion
        Break the graph at a CALL instruction.
```

File: sot/opcode_translator/executor/opcode_executor.py [hunk 6]
```diff
@@ -1093,12 +1374,31 @@ def _prepare_virtual_env(self):
...
+        """
         pycode_gen = PyCodeGen(self._frame)
         fn, inputs = pycode_gen.gen_resume_fn_at(index, stack_size)
         return fn, inputs
 
     @fallback_when_occur_error
-    def _break_graph_in_jump(self, result, instr):
+    def _break_graph_in_jump(self, result: VariableBase, instr: Instruction):
+        """
+        Break the graph at a jump instruction.
```
Comment (Aurelius84) [inline sot/opcode_translator/executor/opcode_executor.py]:
```suggestion
        Break the graph at a JUMP instruction.
```

File: sot/opcode_translator/executor/opcode_executor.py [hunk 7]
```diff
@@ -265,28 +403,74 @@ def __init__(self, code: types.CodeType, graph: FunctionGraph):
...
+        Raises:
+            NotImplementedError: If the method is not implemented.
+
+        """
+        raise NotImplementedError("Please implement virtual_env.")
 
-    def _break_graph_in_jump(self, result, instr):
+    def _break_graph_in_jump(self, result, instr: Instruction):
+        """
+        Breaks the graph in jump instructions.
```
Comment (Aurelius84) [inline sot/opcode_translator/executor/opcode_executor.py]:
```suggestion
        Breaks the graph in JUMP instructions.
```

Comment (zrr1999) [reply sot/opcode_translator/executor/opcode_executor.py]: 好的，已全部修改

File: sot/opcode_translator/executor/opcode_executor.py [hunk 8]
```diff
@@ -90,6 +90,15 @@ class Stop:
 
 @Singleton
 class InstructionTranslatorCache:
+    """
+    A singleton class that implements a cache for translated instructions.
+    This cache is used to store previously translated instructions along with their corresponding guard functions.
+
+    Attributes:
+        cache (dict): A dictionary that maps code objects to tuples of a cache getter function and a list of guarded functions.
+        translate_count (int): The count of how many instructions have been translated.
```
Comment (SigureMo) [inline sot/opcode_translator/executor/opcode_executor.py]:
```suggestion
        translate_count (int): The count of how many instructions have been translated. It is used to test whether the cache hits.
```

**Review by SigureMo — APPROVED** [review]
LGTM～

**Review by Aurelius84 — APPROVED** [review]

**Pull request merged**